features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/async/SnapshotRecordedDataQueueItem.kt

```diff
@@ -16,14 +16,20 @@ internal class SnapshotRecordedDataQueueItem(
     internal val systemInformation: SystemInformation
 ) : RecordedDataQueueItem(recordedQueuedItemContext) {
     internal var nodes = emptyList<Node>()
+    internal var isFinishedTraversal = false
```

Comment:
if this one can be modified from the different threads, it should be either `@Volatile` or `AtomicBoolean`

Reply:
It's modified only on the main thread - created as false and modified to true after the traversal

Reply:
Same may be applied to the `nodes` property as well btw, maybe we should add `@Volatile`/`AtomicReference` there.

Anyway, passing mutable structures between the threads is already a dangerous idea.